### PR TITLE
Fix descomposition of BatchnormInference and update to latest unsqueeze version

### PR DESCRIPTION
**Pull request by roberteg16**

### CODE REVIEW

File: src/Dialect/ONNX/Rewrite.td [hunk 1]
```diff
@@ -69,10 +78,6 @@ def createArrayAttrOf : NativeCodeCall<
 def createArrayAttrOfOneToRankOf : NativeCodeCall<
   "onnx_mlir::createArrayAttrOfOneToN($_builder, $0.getType().cast<ShapedType>().getRank() - 1)">;
 
-// Create an ArrayAttr of IntergerAttr(s) of values in [1, N-2].
```
Comment (mgehre-amd) [inline src/Dialect/ONNX/Rewrite.td:72]: That comment was quite useful. Could you add this and similar comments to the new functions above?

File: test/mlir/onnx/onnx_canonicalization.mlir [hunk 2]
```diff
@@ -117,17 +117,18 @@ func.func @test_conv_batchnormtestmode_fusion_nobias(%arg0: tensor<1x3x224x224xf
...
-    // CHECK:           [[VAR_4_:%.+]] = "onnx.UnsqueezeV11"([[VAR_3_]]) {axes = [1, 2, 3]} : (tensor<*xf32>) -> tensor<*xf32>
-    // CHECK-DAG:       [[VAR_5_:%.+]] = "onnx.Mul"([[PARAM_1_]], [[VAR_4_]]) : (tensor<64x3x7x7xf32>, tensor<*xf32>) -> tensor<*xf32>
-    // CHECK-DAG:       [[VAR_6_:%.+]] = "onnx.Neg"([[PARAM_4_]]) : (tensor<64xf32>) -> tensor<*xf32>
-    // CHECK:           [[VAR_7_:%.+]] = "onnx.Mul"([[VAR_3_]], [[VAR_6_]]) : (tensor<*xf32>, tensor<*xf32>) -> tensor<*xf32>
-    // CHECK:           [[VAR_8_:%.+]] = "onnx.Add"([[PARAM_3_]], [[VAR_7_]]) : (tensor<64xf32>, tensor<*xf32>) -> tensor<*xf32>
-    // CHECK:           [[VAR_9_:%.+]] = "onnx.Conv"([[PARAM_0_]], [[VAR_5_]], [[VAR_8_]]) {auto_pad = "NOTSET", dilations = [1, 1], group = 1 : si64, kernel_shape = [7, 7], pads = [3, 3, 3, 3], strides = [2, 2]} : (tensor<1x3x224x224xf32>, tensor<*xf32>, tensor<*xf32>) -> tensor<1x64x112x112xf32>
+    // CHECK:           [[VAR_1_:%.+]] = onnx.Constant dense<[1, 2, 3]> : tensor<3xi64>
+    // CHECK:           [[VAR_2_:%.+]] = "onnx.Add"([[PARAM_5_]], [[VAR_0_]]) : (tensor<64xf32>, tensor<1xf32>) -> tensor<*xf32>
```
Comment (mgehre-amd) [inline test/mlir/onnx/onnx_canonicalization.mlir:121]: Why does this not have an full result shape?

**Review by mgehre-amd — APPROVED** [review]
Please look into fixing the result shape

File: src/Dialect/ONNX/Rewrite.td (hunk 1 above)
Comment (roberteg16) [reply src/Dialect/ONNX/Rewrite.td:72]: Done

File: test/mlir/onnx/onnx_canonicalization.mlir (hunk 2 above)
Comment (roberteg16) [reply test/mlir/onnx/onnx_canonicalization.mlir:121]: This is all over the rest of the tests too, maybe I have to enable an additional flag. I'll have a look

**Pull request merged**